### ISSUE TITLE
[Feature request] Avoid AOE or dangerous zones

### ISSUE THREAD

**Issue by forest0xia** [forest0xia/dota2bot-OpenHyperAI#47]
Bots cannot see any such AOE like Jakiro's ult or dangerous zones like dark seer's ult. But the api GetAvoidanceZones for this is broken as mentioned in https://www.reddit.com/r/DotA2/comments/1ezxpav. We need a workaround for this

**Comment by christiandwi** [comment]
Confirmed, bot doesn't move when player use upheaval on them

I think all AOE skills always has modifier on the affected enemy (CMIIW), do you mean bot should be able to predict and dodge the AOE before skills are applied?

**Comment by forest0xia** [comment]
>  bot doesn't move when player use upheaval on them
This is not true, I have added a workaround for my bots to retreat from upheaval if they are not full HP and there is no other high desire work they need to do.

> all AOE skills always has modifier
This is not true. I know some skills dont, like Nature's Prophet's Sprout. 

And this GetAvoidanceZones api isn't just for AOEs, ideally bots shouldnt not walk into such AvoidanceZones if there are existing or incoming AOEs (like Lina stun, Kunkka first spell/ult, etc with delays), strong enemies (e.g. currently a bot will retreat directly to base from a lane despite there may be 5 enemies there on the way that are even closer to the base), traps, etc.

As I mentioned, even with modifier, you can't effective tell where the AOE is. bots will walk around the area back and forth if we just rely on modifiers - detect modifier then retreat, no modifier then continue go forward. This results in the bots keep doing dumb things around the AOE location, and easily killed by enemies around with some strategy.
Some AOE last really long or even infinity just like dark seer's ult, upheaval, or SK sand storm, especially with some CD reduction items. 

Bots should really just avoid those dangerous area instead of simple retreat/stepforward.
